applications/nrf9160dk_demo/src/cloud_codec/cloud_codec.c

```diff
@@ -0,0 +1,932 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdbool.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <zephyr.h>
+#include <zephyr/types.h>
+#include <net/cloud.h>
+#include <sys/util.h>
+
+#include "cJSON.h"
+#include "cJSON_os.h"
+#include "cloud_codec.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(cloud_codec, CONFIG_NRF9160DK_DEMO_LOG_LEVEL);
+
+#define CMD_GROUP_KEY_STR "messageType"
+#define CMD_CHAN_KEY_STR "appId"
+#define CMD_DATA_TYPE_KEY_STR "data"
+
+#define DISABLE_SEND_INTERVAL_VAL 0
+#define MIN_INTERVAL_VAL_SECONDS 5
+
+struct cmd {
+	const char *const key;
+	union {
+		enum cloud_cmd_group group;
+		enum cloud_channel channel;
+		enum cloud_cmd_type type;
+	};
+
+	struct cmd *children;
+	size_t num_children;
+};
+
+enum sensor_chan_cfg_item_type {
+	SENSOR_CHAN_CFG_ITEM_TYPE__BEGIN,
+
+	SENSOR_CHAN_CFG_ITEM_TYPE_SEND_ENABLE =
+		SENSOR_CHAN_CFG_ITEM_TYPE__BEGIN,
+	SENSOR_CHAN_CFG_ITEM_TYPE_THRESH_LOW_VALUE,
+	SENSOR_CHAN_CFG_ITEM_TYPE_THRESH_LOW_ENABLE,
+	SENSOR_CHAN_CFG_ITEM_TYPE_THRESH_HIGH_VALUE,
+	SENSOR_CHAN_CFG_ITEM_TYPE_THRESH_HIGH_ENABLE,
+
+	SENSOR_CHAN_CFG_ITEM_TYPE__END
+};
+
+struct sensor_chan_cfg {
+	double value[SENSOR_CHAN_CFG_ITEM_TYPE__END];
+};
+
+struct cloud_sensor_chan_cfg {
+	enum cloud_channel chan;
+	struct sensor_chan_cfg cfg;
+};
+
+#define CMD_ARRAY(...) ((struct cmd[]){ __VA_ARGS__ })
+
+#define CMD_NEW_TYPE(_type) \
+	{ \
+		.key = CMD_DATA_TYPE_KEY_STR, .type = _type, \
+	}
+
+#define CMD_NEW_CHAN(_chan, _children) \
+	{ \
+		.key = CMD_CHAN_KEY_STR, .channel = _chan, \
+		.children = _children, .num_children = ARRAY_SIZE(_children), \
+	}
+
+#define CMD_NEW_GROUP(_var_name, _group, _children) \
+	struct cmd _var_name = { \
+		.key = CMD_GROUP_KEY_STR, \
+		.group = _group, \
+		.children = _children, \
+		.num_children = ARRAY_SIZE(_children), \
+	}
+
+static CMD_NEW_GROUP(group_cfg_set, CLOUD_CMD_GROUP_CFG_SET, CMD_ARRAY(
```

Comment:
remove the items that aren't going to be used in the DK demo from this struct (and related sensor stuff elsewhere in the local codec files)

Reply:
Good idea.

Reply:
Note that I wasn't thoroughly scrubbing these files, since the plan was, at one point, to redo this on top of the event manager architecture that @jhn-nordic is working on.